### DISCUSSION TITLE
Your models in app(s): 'babybuddy' have changes that are not yet reflected in a migration, and so won't be applied.

### DISCUSSION THREAD

**Discussion by flusteredfognozzle** [babybuddy/babybuddy#655] [General]
I am running BabyBuddy v1.16.0 on Docker (Image: lscr.io/linuxserver/babybuddy:latest). On startup, I get the messages:

```
2023-06-02T12:17:49.434454067Z Operations to perform:
2023-06-02T12:17:49.434532283Z   Apply all migrations: admin, auth, authtoken, axes, babybuddy, contenttypes, core, sessions
2023-06-02T12:17:49.434543274Z Running migrations:
2023-06-02T12:17:49.434551072Z   No migrations to apply.
2023-06-02T12:17:49.434583661Z   Your models in app(s): 'babybuddy' have changes that are not yet reflected in a migration, and so won't be applied.
2023-06-02T12:17:49.434592953Z   Run 'manage.py makemigrations' to make new migrations, and then re-run 'manage.py migrate' to apply them.
```

However, I can't seem to get the migrations to work:
```
# docker exec -it babybuddy /bin/bash
# cd /app/www/public/
# python manage.py makemigrations
[...]
ModuleNotFoundError: No module named 'babybuddy.settings.production'
# export DJANGO_SETTINGS_MODULE=babybuddy.settings
# python manage.py makemigrations
No changes detected
# python manage.py migrate
[...]
django.core.exceptions.ImproperlyConfigured: settings.DATABASES is improperly configured. Please supply the ENGINE value. Check settings documentation for more details.
# export DJANGO_SETTINGS_MODULE=babybuddy.settings.base
# python manage.py migrate
[...]
django.core.exceptions.ImproperlyConfigured: The SECRET_KEY setting must not be empty.
```
Having no clue about Django probably prevents me from doing anything towards a solution :)

Patrick

**Comment by cdubz** [comment]
👋 yeah this is benign. It has to do with how Baby Buddy generates the list of time zone options. It’s a silly thing I should fix but just haven’t yet. Nothing to worry about!

**Comment by flusteredfognozzle** [comment]
Ah, no problem, nothing to solve. Perfect. Anyways thank you for the great application. My wife loves it :)

**Comment by danepowell** [comment]
> However, I can't seem to get the migrations to work:

Just FYI, the Docker image automatically runs migrations, I believe on restart: https://github.com/linuxserver/docker-babybuddy/blob/32493cc0b2d885e814e0d6b73e6291461f32bdf9/root/etc/s6-overlay/s6-rc.d/init-babybuddy-config/run#L23

The message here isn't indicating that you need to run migrations, but rather the app has some discrepancy that needs to be _captured_ in a new migration.

I was also confused / alarmed by this, but if Chris says it's okay, I'll try to find something more important to worry about 🙈
